com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/EvaluateRequestHandler.java

```diff
@@ -54,130 +54,78 @@
-                    ErrorCode.EVALUATE_FAILURE,
+        StackFrameReference stackFrameReference = (StackFrameReference) context.getRecyclableIdPool().getObjectById(evalArguments.frameId);
+        if (stackFrameReference == null) {
+            // stackFrameReference is null means the stackframe is continued by user manually,
+            return AdapterUtils.createAsyncErrorResponse(response, ErrorCode.EVALUATE_FAILURE,
                     "Failed to evaluate. Reason: Cannot evaluate because the thread is resumed.");
         }
 
-        // split a.b.c => ["a", "b", "c"]
-        List<String> referenceExpressions = Arrays.stream(StringUtils.split(expression, '.'))
-                .filter(StringUtils::isNotBlank).map(StringUtils::trim).collect(Collectors.toList());
-
-        // get first level of value from stack frame
-        Variable firstLevelValue = null;
-        boolean inStaticMethod = stackFrameProxy.getProxiedObject().location().method().isStatic();
-        String firstExpression = referenceExpressions.get(0);
-        // handle special case of 'this'
-        if (firstExpression.equals("this") && !inStaticMethod) {
-            firstLevelValue = VariableUtils.getThisVariable(stackFrameProxy.getProxiedObject());
-        }
-        if (firstLevelValue == null) {
-            try {
-                // local variables first, that means
-                // if both local variable and static variable are found, use local variable
-                List<Variable> localVariables = VariableUtils.listLocalVariables(stackFrameProxy.getProxiedObject());
-                List<Variable> matchedLocal = localVariables.stream()
-                        .filter(localVariable -> localVariable.name.equals(firstExpression)).collect(Collectors.toList());
-                if (!matchedLocal.isEmpty()) {
-                    firstLevelValue = matchedLocal.get(0);
-                } else {
-                    List<Variable> staticVariables = VariableUtils.listStaticVariables(stackFrameProxy.getProxiedObject());
-                    List<Variable> matchedStatic = staticVariables.stream()
-                            .filter(staticVariable -> staticVariable.name.equals(firstExpression)).collect(Collectors.toList());
-                    if (matchedStatic.isEmpty()) {
-                        return AdapterUtils.createAsyncErrorResponse(response,
-                                ErrorCode.EVALUATE_FAILURE,
-                                String.format("Failed to evaluate. Reason: Cannot find the variable: %s.", referenceExpressions.get(0)));
-                    }
-                    firstLevelValue = matchedStatic.get(0);
-                }
-
-            } catch (AbsentInformationException e) {
-                // ignore
+        CompletableFuture<Response> completableFuture = new CompletableFuture<>();
+        // new thread to unsure that the logic to acquire locker and release locker will be executed in this thread
+        new Thread(() -> {
```

Comment:
Try not start a new thread this way. We need to let the executor run the thread. Here we can use the Async version of future methods to schedule all actions using the default executor.